srml/session/src/lib.rs

```diff
@@ -490,6 +490,9 @@ impl<T: Trait> Module<T> {
 
 		let applied_at = session_index + 2;
 
+		// Inform the session handlers that a session is going to end.
+		T::SessionHandler::on_before_session_ending();
```

Comment:
Should we call this before `Validators<T>` is populated with new validators? I'd assume that the state still describes the previous session.